just_audio/lib/just_audio.dart

```diff
@@ -1254,6 +1255,11 @@ class AudioPlayer {
         }
       }
       if (_disposed) return _platform;
+
+      if (_isLibWinMedia()) {
+        LibWinMediaJustAudioPlugin.registerWith();
+      }
```

Comment:
Override the current platform interface with `LibWinMediaJustAudioPlugin` whenever we are running on Windows or Linux!

Reply:
For now let's do something similar to `just_audio_background` where the dependency is not automatically endorsed but rather the app developer adds the dependency themselves and calls the `registerWith` method themselves (the latter step hopefully being automated in a future version of Flutter, because it seems they did fix this in a recent PR).

Reply:
But we don't use platform code here, only FFI. There's no way for flutter to register Windows or Linux automatically.

We can take the official path_provider plugin as example:

https://github.com/flutter/plugins/blob/master/packages/path_provider/path_provider/lib/path_provider.dart#L44-L61

Windows and Linux requires manual registration.

Reply:
Yes but as I mentioned above there was recent work on flutter to rectify this. See https://github.com/flutter/flutter/issues/52267